src/common/mmap.h

```diff
@@ -107,6 +110,39 @@ util_map_hint_align(size_t len, size_t req_align)
 	return align;
 }
 
+typedef enum map_tracker_flags {
+	MTF_DIRECT_MAPPED = 0x0001,
+
+	/* this should hold the value of all flags ORed for debug purpose */
+	MTF_MASK = MTF_DIRECT_MAPPED
+} map_tracker_flags;
+
+/*
+ * this structure tracks the file mappings outstanding per file handle
+ */
+typedef struct map_tracker {
+	SORTEDQ_ENTRY(map_tracker) entry;
+	const void *base_addr;
+	const void *end_addr;
+	map_tracker_flags flags;
+
+#ifdef _WIN32
+	/* Windows-specific data */
+	HANDLE FileHandle;
+	HANDLE FileMappingHandle;
+	DWORD Access;
+	off_t Offset;
+	size_t FileLen;
+#endif
+} map_tracker;
+
+extern pthread_rwlock_t Mmap_list_lock;
+extern SORTEDQ_HEAD(map_list_head, map_tracker) Mmap_list;
+
+int util_range_track(const void *addr, size_t len);
+int util_range_untrack(const void *addr, size_t len);
+map_tracker *util_range_find(const void *addr, size_t len);
```

Comment:
I think this public interface shouldn't expose the map_tracker structure, but instead return a bool whether or not the provided range is "tracked". That would allow you to hide all of the structures and remove the extern variables.